ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -460,12 +465,25 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
 
       final Schema schema =
           SchemaUtil.resolveBinaryOperatorResultType(
-              left.getRight().type(), right.getRight().type());
+              left.getRight(), right.getRight(), node.getOperand());
 
-      return new Pair<>(
-          "(" + left.getLeft() + " " + node.getType().getValue() + " " + right.getLeft() + ")",
-          schema
-      );
+      if (DecimalUtil.isDecimal(schema)) {
+        return new Pair<>(
+            String.format(
```

Comment:
Couldn't find tests for this. Are you adding tests for decimal?

Reply:
These are all tested in the JSON files, I will add SqlToJavaVisitorTest